### PR TITLE
MRG: Test warnings if NIRS channels are not correctly ordered

### PR DESCRIPTION
**Pull request by rob-luke**
#### Reference issue
General fNIRS tweaking #7057 

#### What does this implement/fix?
Added test to ensure that NIRS data is stored as paired wavelength data. Noticed that if an odd number of channels existed that the check failed.

### CODE REVIEW

**Comment by rob-luke** [comment]
@larsoner @agramfort could you please review this. Thanks

The failing CI test seems unrelated

File: mne/preprocessing/nirs/nirs.py [hunk 1]
```diff
@@ -70,17 +70,20 @@ def _check_channels_ordered(raw, freqs):
...
+            ch2_name_info = re.match(r'S(\d+)_D(\d+) (\d+)',
+                                     raw.info['chs'][ii + 1]['ch_name'])
+
+            if (ch1_name_info.groups()[0] != ch2_name_info.groups()[0]) or \
+               (ch1_name_info.groups()[1] != ch2_name_info.groups()[1]) or \
+               (int(ch1_name_info.groups()[2]) != freqs[0]) or \
+               (int(ch2_name_info.groups()[2]) != freqs[1]):
+                raise RuntimeError('NIRS channels not ordered correctly')
+    else:
+        raise RuntimeError('NIRS channels not ordered correctly')
```
Comment (agramfort) [inline mne/preprocessing/nirs/nirs.py]: I would have used a ValueError and I would give more details on the error eg printing the problematic values. YOu could also avoid copy pasting the error message. besides LGTM

**Comment by rob-luke** [comment]
Thanks @agramfort. Sorry but I tweaked the error messages again after you approved the PR. I tried to better match other code in MNE.

**Comment by larsoner** [comment]
Pushed a cosmit to reduce the indentation level, will merge once CIs come back happy -- thanks @rob-luke

**Pull request merged**